### PR TITLE
Fix localization with multiple correct answers in Long Menu question as Page content (Mantis #35064) (release_6)

### PR DESCRIPTION
**Pull request by nicoroeser**
This is a fix for [Mantis #35064](https://mantis.ilias.de/view.php?id=35064 "Mantis #35064: Failed test: Langes-Menü-Frage erstellen/bearbeiten"). See the commit message for details.

This is the pull request for the `release_6` branch. Other pull requests are opened for other branches.

@Uwe-Kohnle , can you please have a look at this? ILIAS 6 and ILIAS 7 still reference `question_handling.js` in the Scorm2004 code, but I have not found a location in the code which would require a change like the one for `class.ilPCQuestion.php` in this pull request. If there is none, fine. If there is, please point me to it so I can add the fix accordingly before this is merged.

### CODE REVIEW

**Comment by dsstrassner** [comment]
Hi @mbecker-databay & @kergomard & @Uwe-Kohnle, 

[Mantis #35064](https://mantis.ilias.de/view.php?id=35064) is in Mantis Category "Learning Module ILIAS : Assessment Questions". 
The Fix would move  [Testcase 54937](http://testrail.ilias.de/index.php?/tests/view/54937) to fully passed. (PR #5238). 
Should this PR be merged by @Uwe-Kohnle or @alex40724 or someone from the T&A?

I will cross post this comment in all PRs regarding this Mantis issue.

**Comment by kergomard** [comment]
Thank you very much for the PRs @nicoroeser . I think this is for you @alex40724 . I will deassign Max and me. From where I stand this looks good and can be merged.

Thanks also to you @alex40724 .

**Comment by alex40724** [comment]
@nicoroeser Thanks for the PR. In general it is not necessary to open PRs for all branches. Maintainers are responsible to take care of all necessary merges. To have all these PRs may look convenient for developers first, but since PRs come with discussions and assignments it quickly becomes unpractical to do these redundantly. So I will not copy my comments, just do it here in the oldest reported version.

@dsstrassner The history of the js file is quite complex. Originally it has been implemented by Hendrik Holtmann at the ISN as part of the SCORM editor. When the ISN stopped the maintenance, Leifos took over the SCORM editor maintenance. However this file has been maintained/changed by Björn a lot and mainly Databay maintained it, since the code is highly coupled to the T&A code. Features like the Long Menu Question (Guido) have been added and since then most fixes are by Michael and Björn.

I am currently not ready to take over the maintenance of this file completely, since I am missing a defined interface to the T&A code. So in my opinion this should be at least reviewed by the T&A maintainer before merge, since I have limited knowledge, e.g. of the Long Menu Question implementation. I can merge this after a review however.

**Review by alex40724 — APPROVED** [review]

**Comment by alex40724** [comment]
@dsstrassner I added @mbecker-databay as reviewer, you may change this to @gvollbach since he integrated the long menu question related code. Or any other T&A expert who might review this.

**Review by kergomard — APPROVED** [review]
I've coordinated with @dsstrassner and this can be merged. Thanks!

**Comment by dsstrassner** [comment]
As @kergomard says, this can be merged.

**Comment by alex40724** [comment]
Merged to ILIAS 7, 8, trunk

**Comment by nicoroeser** [comment]
Thank you, @alex40724, for handling this!

The code change is integrated in `release_6`, `release_7` and `trunk` now, but the merge into `release_8` seems to have been dropped. **Could you please look into this once more? Thanks again!**

And unfortunately the commit message (which contains explanations about the changes and for possible future steps) is only available on the branch for ILIAS 6. This is a bit inconvenient because further work either needs this pull request (and an online connection) or requires searching for the same change on all branches (which will normally not been done during e.g. a bisect operation or when just reading the output of `git log trunk -- Modules/Scorm2004/` and the like).

I understand your [note](https://github.com/ILIAS-eLearning/ILIAS/pull/5236#issuecomment-1332028989) about not creating pull requests for each branch (thanks for the explanation!). When porting one or more commits to another branch, I’d suggest to use cherry-picking in order not to lose the commit history. In case of a _series_ of commits instead of a single one, some prefer merging with `--no-ff` (but every project handles things differently of course).

**Pull request merged**
